### PR TITLE
Adding VIA support for CannonKeys/Practice60

### PR DESCRIPTION
**Pull request by morbetter**
## Description

Adding VIA support to cannonkeys/practice60

* updated VENDOR_ID to match other CannonKeys boards
* changed PRODUCT_ID to be unique
* added additional notes to readme.md
* keymap.c and config.h for VIA support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by awkannan** [comment]
@zvecr Did we ever add eeprom support for f103? If not, this won't really work.

**Comment by morbetter** [comment]
I was using VIA this afternoon to change the layout on my Practice60, for whatever that's worth.

**Comment by zvecr** [comment]
@awkannan Theres some support <https://github.com/qmk/qmk_firmware/blob/master/common_features.mk#L137>, but i know it gets super unhappy on fake chips, #7204.

File: keyboards/cannonkeys/practice60/readme.md [hunk 1]
```diff
@@ -10,3 +10,13 @@ Make example for this keyboard (after setting up your build environment):
...
 
 See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+Build guide available at: https://docs.cannonkeys.com/bluepill-build/
+Blue Pill flashing guide at: https://docs.cannonkeys.com/flashing/
+Flash using dfu-util:
+
+    dfu-util.exe -d 1eaf:0003 -a 2 -D "\path\to\firmware.bin"
+
+Note: If using MSYS2 and getting AVR-GCC errors try running the command below and restarting MSYS2. See this link for more information: https://github.com/qmk/qmk_firmware/issues/4318
```
Comment (fauxpark) [inline keyboards/cannonkeys/practice60/readme.md]: Firmware for Arm chips isn't compiled with avr-gcc. I'm not sure what this has to do with this board.

Comment (morbetter) [reply keyboards/cannonkeys/practice60/readme.md]: Apologies, I copy/pasted the wrong thing.  That link was what led me to the script to fix the errors though.

**Comment by morbetter** [comment]
Bump?

File: keyboards/cannonkeys/practice60/readme.md [hunk 2]
```diff
@@ -10,3 +10,13 @@ Make example for this keyboard (after setting up your build environment):
     make cannonkeys/practice60:default
```
Comment (fauxpark) [inline keyboards/cannonkeys/practice60/readme.md:10]:
```suggestion
    make cannonkeys/practice60:default

Flashing example for this keyboard:

    make cannonkeys/practice60:default:flash
```

File: keyboards/cannonkeys/practice60/readme.md [hunk 3]
```diff
@@ -10,3 +10,13 @@ Make example for this keyboard (after setting up your build environment):
...
+
+Build guide available at: https://docs.cannonkeys.com/bluepill-build/
+Blue Pill flashing guide at: https://docs.cannonkeys.com/flashing/
+Flash using dfu-util:
+
+    dfu-util.exe -d 1eaf:0003 -a 2 -D "\path\to\firmware.bin"
+
+Note: If using MSYS2 and getting gcc-arm-none-eabi errors try running the command below and restarting MSYS2.
+
+    echo "source ~/qmk_utils/activate_msys2.sh" >> ~/.bashrc
```
Comment (fauxpark) [inline keyboards/cannonkeys/practice60/readme.md:22]:
```suggestion
```

File: keyboards/cannonkeys/practice60/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/cannonkeys/practice60/keymaps/via/rules.mk:2]: To be honest, LTO isn't really needed here.  It's present on the AVR boards due to firmware size limitations, and I don't think that the F103 has the same issue.  upas/awkannan or somebody more knowledgeable about the hardware would have to confirm.

File: keyboards/cannonkeys/practice60/config.h [hunk 5]
```diff
@@ -18,8 +18,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x6464
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x5060
```
Comment (zvecr) [inline keyboards/cannonkeys/practice60/config.h:22]: given `practice65` is 6565,
```suggestion
#define PRODUCT_ID      0x6060
```

File: keyboards/cannonkeys/practice60/config.h [hunk 6]
```diff
@@ -18,8 +18,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x6464
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x6060
 #define DEVICE_VER      0x0001
 /* in python2: list(u"whatever".encode('utf-16-le')) */
 /*   at most 32 characters or the ugly hack in usb_main.c borks */
```
Comment (Erovia) [inline keyboards/cannonkeys/practice60/config.h:25]:
```suggestion

```

File: keyboards/cannonkeys/practice60/config.h [hunk 7]
```diff
@@ -18,8 +18,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x6464
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x6060
 #define DEVICE_VER      0x0001
```
Comment (Erovia) [inline keyboards/cannonkeys/practice60/config.h:23]: Sadly github does not allow me to comment on that line, but could you change Line25 to:
`#define MANUFACTURER CannonKeys`

**Comment by morbetter** [comment]
@Erovia, I tried that, but there didn't appear to be any changes after updating from master.

> Also, I'm not sure what 294901a does in this PR.

Where was this mentioned?

**Comment by Erovia** [comment]
It's the very first commit in this PR.

**Comment by morbetter** [comment]
Well that's odd.  I don't know how that ended up in there, and I'm not sure how to remove it.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**